snoop/data/digests.py

```diff
@@ -170,13 +175,30 @@ def gather(blob, **depends_on):
     with models.Blob.create() as writer:
         writer.write(json.dumps(rv).encode('utf-8'))
 
-    _, _ = models.Digest.objects.update_or_create(
+    digest_obj, _ = models.Digest.objects.update_or_create(
         blob=blob,
         defaults=dict(
             result=writer.blob,
         ),
     )
 
+    # get thumbnail and create new thumbnail entry in the database
+    thumbnail_blobs = depends_on.get('get_thumbnail')
+
+    if thumbnail_blobs:
+        if isinstance(thumbnail_blobs, SnoopTaskBroken):
+            rv['broken'].append(thumbnail_blobs.reason)
+            log.debug("get_thumbnail task is broken; skipping")
```

Comment:
Also add a boolean flag here called `has-thumbnails` that is `False` unless otherwise set.